homework/index.js

```diff
@@ -30,13 +30,149 @@
+          );
+
+          // Call another function(addContributors) to fill i contributor information.
+          fetchJSON(repoInfo[repo].contributors_url, (err, data) => {
+            const root = document.getElementById('root');
+
+            if (err) {
+              createAndAppend('div', root, { text: err.message, class: 'alert-error' });
+            } else {
+              addContributors(data);
+            }
+          });
+        }
+      }
+    }
+    const htmlString = templateElement.join('');
+    document.getElementById('section1').innerHTML = htmlString;
+  }
+
+  // Create options under 'SELECT' element which should have HYF Repositories.
+
+  function loadSelectionValues(userInfo) {
+    const sortRepoName = [];
+
+    const userRepo = JSON.parse(userInfo);
+    const mySelect = document.getElementById('myselect');
+
+    // push all the HYP repo names to sort array.
+
+    // eslint-disable-next-line no-restricted-syntax
+    for (const repo in userRepo) {
+      if (Object.prototype.hasOwnProperty.call(userRepo, repo)) {
+        sortRepoName.push(userRepo[repo].name);
+      }
+    }
+
+    // sort the repo name using sort function and localeComapare for uppercase and lowercase sorting.
+    sortRepoName.sort((a, b) => a.localeCompare(b));
+
+    // Create Option under Select element and attach the  same with SELECT element.
+
+    // eslint-disable-next-line no-restricted-syntax
+    for (const repo of sortRepoName) {
+      const option = document.createElement('option');
+      option.value = repo;
+      option.text = repo;
+      mySelect.appendChild(option);
+    }
+
+    const selectBox = document.getElementById('myselect');
+    const selectedValue = selectBox.options[selectBox.selectedIndex].value;
+
+    // Load Repository information for the choose repository name in the select box.
+    loadDiv(userRepo, selectedValue);
+    selectBox.onchange = function() {
+      loadDiv(userRepo, selectBox.value);
+    };
+  }
+
   function main(url) {
     fetchJSON(url, (err, data) => {
       const root = document.getElementById('root');
+      const myParent = document.body;
+
       if (err) {
         createAndAppend('div', root, { text: err.message, class: 'alert-error' });
       } else {
-        createAndAppend('pre', root, { text: JSON.stringify(data, null, 2) });
+        // Create div element 'select' in  document body to hold the label element and list box.
+
+        createAndAppend('div', myParent, { id: 'select' });
+        const select = document.getElementById('select');
+        createAndAppend('LABEL', select, {
+          text: 'HYF Repositories: ',
+          id: 'label',
+          for: 'repo',
+        });
+        createAndAppend('select', select, { id: 'myselect' });
+
+        // Create two div elements section1 and section2 under 'Root' div to have
+        // section1 - Repository Information.
+        // section2 - Contributions.
+
+        createAndAppend('div', root, { id: 'section1' });
```

Comment:
Would name these divs closer to what content they will hold. More like `contributors` and `repo-details`

Reply:
corrected the name div name as repodetails,contributorinfomation